subprojects/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -221,6 +233,42 @@ private static void validatePackageName(String packageName) {
         }
     }
 
+    static class JavaVersionSelection {
+        private final JavaLanguageVersion languageVersion;
+
+        JavaVersionSelection(JavaLanguageVersion ver) {
+            languageVersion = ver;
+        }
+
+        @Override
+        public String toString() {
+            return "Java " + languageVersion;
+        }
+
+        public JavaLanguageVersion getLanguageVersion() {
+            return languageVersion;
+        }
+    }
+
+    java.util.Optional<JavaLanguageVersion> getJavaLanguageVersion(boolean useIncubatingAPIs, UserInputHandler inputHandler, Language language) {
+        if (isJvmLanguage(language) && useIncubatingAPIs) {
```

Comment:
Toolchain selection/provisioning is not considered incubating now, so I think it's OK to always ask this question vs looking at that flag.  The "use incubating API" flag was intended for using incubating features in the generated build and not introducing incubating behaviors to the init process.